storage/opsmgr.go

```diff
@@ -154,34 +166,130 @@ func (v *OmAccessor) Get(path string) (map[string]string, error) {
+	propertyRaw := strings.Split(property, ".")
+	if len(propertyRaw) != 3 {
+		return map[string]string{}, fmt.Errorf("property '%s' is not properly formatted", property)
+	}
+	property = propertyRaw[2]
+
+	var settings struct {
+		Products []map[string]interface{} `json:"products"`
+	}
 
-		err = json.Unmarshal(respBody, &credentialReferences)
-		if err != nil {
-			return []string{}, fmt.Errorf("could not unmarshal credentials response: %s\nresponse: `%s`", err, respBody)
+	respBody, err := v.opsmanAPI("/api/installation_settings")
+	if err != nil {
+		return map[string]string{}, err
+	}
+
+	err = json.Unmarshal(respBody, &settings)
+	if err != nil {
+		return map[string]string{}, fmt.Errorf("could not unmarshal certificates response: %s\nresponse: `%s`", err, respBody)
+	}
+
+	for _, product := range settings.Products {
+		if product["guid"] == guid {
+			propertyMap, isMap := product[property].(map[string]interface{})
+			if !isMap {
+				return map[string]string{}, fmt.Errorf("failed to unmarshal property %s from product in fallback api", property)
+			}
+			key, isMap := propertyMap["cert_pem"].(string)
+			if !isMap {
+				return map[string]string{}, fmt.Errorf("failed to unmarshal cert from property %s in fallback api", property)
+			}
+			return map[string]string{
+				"cert_pem": key,
+			}, nil
 		}
+	}
+
+	return map[string]string{}, nil
+}
 
-		for _, cred := range credentialReferences.Credentials {
-			finalPaths = append(finalPaths, fmt.Sprintf("/api/v0/deployed/products/%s/credentials/%s", deployment, cred))
+func (v *OmAccessor) opsmanRootFallback(guid string, property string) (map[string]string, error) {
+	propertyRaw := strings.Split(property, ".")
+	if len(propertyRaw) != 4 {
+		return map[string]string{}, fmt.Errorf("property '%s' is not properly formatted", property)
+	}
+	property = propertyRaw[2]
+	certGUID := propertyRaw[3]
+
+	var certificateAuthorities struct {
+		CAs []struct {
+			GUID        string    `json:"guid"`
+			Issuer      string    `json:"issuer"`
+			CreatedOn   time.Time `json:"created_on"`
+			ExpiresOn   time.Time `json:"expires_on"`
+			Active      bool      `json:"active"`
+			CertPem     string    `json:"cert_pem"`
+			NatsCertPem string    `json:"nats_cert_pem"`
+		} `json:"certificate_authorities"`
+	}
+
+	respBody, err := v.opsmanAPI("/api/v0/certificate_authorities")
+	if err != nil {
+		return map[string]string{}, err
+	}
+
+	err = json.Unmarshal(respBody, &certificateAuthorities)
+	if err != nil {
+		return map[string]string{}, fmt.Errorf("could not unmarshal certificates response: %s\nresponse: `%s`", err, respBody)
+	}
+
+	for _, ca := range certificateAuthorities.CAs {
+		if ca.GUID == certGUID {
+			if property == "nats_client_ca" {
+				return map[string]string{
+					"cert_pem": ca.NatsCertPem,
+				}, nil
+			}
+			if property == "root_ca" {
+				return map[string]string{
+					"cert_pem": ca.CertPem,
+				}, nil
+			}
```

Comment:
is there a reason we would only want to look for these two CAs, even if the GUID matches? Would the following be more robust?
```suggestion
			if property == "nats_client_ca" {
			    ca.CertPem = ca.NatsCertPem
			}
			
			return map[string]string{"cert_pem": ca.CertPem}, nil
```

Reply:
In this case they're equivalent as the `/api/v0/certificate_authorities` endpoint only returns the root and nats ca for each authority and if they added additional certs to the endpoint we'd have to be grabbing them by a different key in the json structure. 

ex. output from that api
```
{
  "certificate_authorities": [
    {
      "guid": "58649b43a49c62a85a47",
      "issuer": "Pivotal",
      "created_on": "2019-05-09T14:35:15Z",
      "expires_on": "2023-05-09T14:35:15Z",
      "active": true,
      "cert_pem": "-----BEGIN CERTIFICATE-----\n...\n-----END CERTIFICATE-----\n",
      "nats_cert_pem": "-----BEGIN CERTIFICATE-----\n...\n-----END CERTIFICATE-----\n"
    },
    {
      "guid": "064d734711f003e8f565",
      "issuer": "Pivotal",
      "created_on": "2019-12-23T15:11:41Z",
      "expires_on": "2023-12-23T15:11:41Z",
      "active": false,
      "cert_pem": "-----BEGIN CERTIFICATE-----\n...\n-----END CERTIFICATE-----\n",
      "nats_cert_pem": "-----BEGIN CERTIFICATE-----\n...\n-----END CERTIFICATE-----\n"
    }
  ]
}
```